lib/service_actor/result.rb

```diff
@@ -1,16 +1,19 @@
 # frozen_string_literal: true
 
-require "ostruct"
-
 # Represents the context of an actor, holding the data from both its inputs
 # and outputs.
-class ServiceActor::Result
+
+class ServiceActor::Result < BasicObject
   def self.to_result(data)
     return data if data.is_a?(self)
 
     new(data.to_h)
   end
 
+  %i[class respond_to? is_a? kind_of? send].each do |mid|
```

Comment:
Is `respond_to?` needed here if we override it in the same file?

```suggestion
  %i[class respond_to? is_a? kind_of? send].each do |mid|
```

Reply:
Yep, thanks